### PR TITLE
Qg 9 sign in page

### PR DESCRIPTION
**Pull request by huydoan1908**

### CODE REVIEW

File: src/components/Form/Login.tsx [hunk 1]
```diff
@@ -69,7 +96,21 @@ const Login = () => {
...
+          className={styles.formSubmit}
+          disabled={mutation.isLoading}
+        >
+          {mutation.isLoading && (
+            <Spinner
+              as="span"
+              animation="border"
+              size="sm"
+              role="status"
+              className="me-2"
```
Comment (huy27201) [inline src/components/Form/Login.tsx:113]: Could you please extract the Spinner component and use it in the Loader component? We can pass `...props` to the Loader component. I will use it in my tasks.

File: src/components/Layouts/AppLayout/AppLayout.tsx [hunk 2]
```diff
@@ -1,13 +1,22 @@
 import React from 'react';
+import { useNavigate } from 'react-router-dom';
 import { Container } from 'react-bootstrap';
 import Footer from '../Footer/Footer';
 import Header from '../Header';
+import { isLogin } from '../../../utils/auth';
```
Comment (huy27201) [inline src/components/Layouts/AppLayout/AppLayout.tsx]: Let's create an `index.ts` file to import and export in `utils`.

File: src/components/Layouts/AuthLayout/index.tsx [hunk 3]
```diff
@@ -1,12 +1,21 @@
 import React from 'react';
 import { Container, Button } from 'react-bootstrap';
+import { useNavigate } from 'react-router-dom';
 import styles from './AuthLayout.module.scss';
+import { isLogin } from '../../../utils/auth';
```
Comment (huy27201) [inline src/components/Layouts/AuthLayout/index.tsx]: Same as above

File: src/routes/index.tsx [hunk 4]
```diff
@@ -1,37 +1,24 @@
...
   {
     path: '/',
-    element: <Navigate to="/login" />,
-    errorElement: (
-      <AppLayout>
-        <Loader />
-      </AppLayout>
-    ),
+    element: <HomePage />,
+    errorElement: <Loader />,
```
Comment (huy27201) [inline src/routes/index.tsx:13]: What is `errorElement` used for?

File: src/utils/localStorage.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  const item = expireIn ? { value, expireIn: current } : { value };
+  localStorage.setItem(key, JSON.stringify(item));
+};
+
+export const getItem = (key: string) => {
+  const value = localStorage.getItem(key) || '';
+  try {
+    const item = JSON.parse(value);
+    const expireIn = item?.expireIn;
+    if (expireIn && new Date(expireIn) < new Date()) {
```
Comment (huy27201) [inline src/utils/localStorage.ts]: I think we can extract this Date comparison logic in `utils`

File: src/routes/index.tsx (hunk 4 above)
Comment (huydoan1908) [reply src/routes/index.tsx:13]: like 404 page when go to unknow url e.g.  /1 /2 /3

File: src/utils/localStorage.ts (hunk 5 above)
Comment (huy27201) [reply src/utils/localStorage.ts]: example: `date.util.ts`

**Review by huy27201 — APPROVED** [review]

**Pull request merged**